custom-recipes/text-extraction/recipe.py

```diff
@@ -30,18 +35,31 @@
 
     with input_folder.get_download_stream(sample_file) as stream:
         file_bytes = stream.read()
-
-    try:
-        extracted_text = extract_text_content(file_bytes, suffix, with_pandoc)
-
-        if not extracted_text.strip():
-            logger.warning("Extracted text is empty")
-
-        rows.append({'file': sample_file, 'text': extracted_text, 'error_message': ""})
-        logger.info("Extracted text from {}/{} files (in {:.2f} seconds)".format(i+1, total_files, perf_counter() - start))
-    except Exception as e:
-        rows.append({'file': sample_file, 'text': "", 'error_message': e})
-        logger.info("Failed extracting text from file {} because: {}".format(sample_file, e))
+    
+    if params["extract_chunks"]:
+        try:
+            extracted_chunks = extract_text_chunks(sample_file, file_bytes, suffix, with_pandoc)
+
+            if len(extracted_chunks) == 0:
+                logger.warning("Extracted chunks are empty")
+
+            rows += extracted_chunks
```

Comment:
[nitpickish] We could use `.extend` to avoid creating a new list in memory every time, but should not be too problematic here